MapboxNavigation/InstructionLabel.swift

```diff
@@ -23,32 +26,20 @@ open class InstructionLabel: StylableLabel {
         let string = NSMutableAttributedString()
         
         // Add text or image
-        for component in instruction.components {
-            let isFirst = component == instruction.components.first
+        for component in instruction {
+            let isFirst = component == instruction.first
             let joinChar = !isFirst ? " " : ""
             
-            if let prefix = component.prefix {
-                string.append(NSAttributedString(string: prefix, attributes: attributes))
-            }
-            
-            if let roadCode = component.roadCode, let network = component.network, let number = component.number {
-                // Check if shield image is cached, otherwise display road code in text
-                let shieldKey = UIImage.shieldKey(network, number: number, height: shieldHeight)
+            if let imageURL = component.imageURLS[scale] {
+                let shieldKey = UIImage.shieldKey(imageURL, height: shieldHeight)
                 if let cachedImage = UIImage.cachedShield(shieldKey) {
                     string.append(attributedString(with: cachedImage))
-                    if let direction = component.direction {
-                        string.append(NSAttributedString(string: " "+direction, attributes: attributes))
-                    }
                 } else {
-                    // Download shield and display road code in the meantime
-                    string.append(NSAttributedString(string: joinChar+roadCode, attributes: attributes))
```

Comment:
I think we should keep this stringified version of the shield while the image is being downloaded which also works if the request fails.